coveralls/api.py

```diff
@@ -272,6 +272,13 @@ def git_info():
                 }]
             }
         """
+        # silently omit the optional 'git' API argument if a simple git status
+        # command fails, suggesting git is unavailable
+        try:
+            run_command('git', 'status')
+        except CoverallsException:
+            return {}
+
         rev = run_command('git', 'rev-parse', '--abbrev-ref', 'HEAD').strip()
```

Comment:
Could you please put the try catch around the already-existing commands? No need to run an extra git command here.

Reply:
I thought about that; the original reason I ended up adding a `git status` command was to avoid any possible errors related specifically to the existing commands, that may be because of something other than git not being available.

I can't think of a *good* scenario but an unlikely example could be having a git version that does not support the current command (or any future changes). Either because git version is too old or too new with backwards incompatible cli changes.

WDYT?

Reply:
That makes sense -- I think if we add the logging, something like "could not collect git data -- are you running coveralls in a repository?", along with logging the CoverallsException message (which contains the process output), that should cover it!

Reply:
Yes. Will amend with a more verbose log (and phrasing that's more to your liking).